src/main/java/com/elmsoftware/env/EnvironmentSettingsModule.java

```diff
@@ -55,4 +57,18 @@ protected void configure(final Binder binder, final Map<String, String> properti
 
 	}
 
+	protected void overrideEnvironmentSettings(final EnvironmentSettings environmentSettings, final String environment) {
+		final Map<String, String> settings = environmentSettings.getEnvironmentSettings().get(environment);
+
+		settings.keySet()
+			.forEach(key -> {
+				final String envKey = key.replaceAll("[^A-Za-z0-9]", "_").toUpperCase(Locale.US);
+				final String envValue = System.getenv(envKey);
+				if (envValue != null) {
+					log.debug("Overriding " + key + " with environment variable");
```

Comment:
i originally wrapped this with a debug enabled check:
```suggestion
					if (log.isDebugEnabled()) {
						log.debug("Overriding " + key + " with environment variable");
					}
```
but getting the code coverage for that was going to be annoying. generating that string isn't too expensive at least

Reply:
Use the `{}` format for logging, it's pretty cheap and clean; all it has to do is push the reference onto the stack, then it decides of it needs to build the string or not later.